### PR TITLE
Hlsl support

### PR DESCRIPTION
**Pull request by progDes007**
Pardon my "art". I hope legendary teapot will do :).
image

### CODE REVIEW

File: languages.yaml [hunk 1]
```diff
@@ -1071,6 +1071,38 @@ Hcl:
...
+    {0}   $$$$$$$$$$$$$$$$$$$$$$$$$${1}$$$$$
+    {0}     $$$$$$$$$$$$$$$$$$$${1}$$$$$$$
+    {0}      {1}$$$$$$$$$$$$${1}$$$$$$$$$$$
+    {0}         {1}$$$$$$$$$$$$$$$$$$
+    {2}       _      _  _        _____  _   
+    {2}      |:|    |:||:|      /:::::||:|
+    {2}      |:|____|:||:|     |:(___  |:|
+    {2}      |::::::::||:|      \::::\ |:|
+    {2}      |:|    |:||:|____  ____):||:|____
+    {2}      |:|    |:||::::::||:::::/ |::::::|
```
Comment (spenserblack) [inline languages.yaml:1095]: IMO it looks a bit unusual with borders on the left, right, and top sides of the letters, but not the bottom. Have you considered adding a bottom border?
```
    {2}       _      _  _        _____  _   
    {2}      |:|    |:||:|      /:____||:|
    {2}      |:|____|:||:|     |:(___  |:|
    {2}      |::____::||:|      \___:\ |:|
    {2}      |:|    |:||:|____  ____):||:|____
    {2}      |_|    |_||______||_____/ |______|
```

This is just a basic idea. It sacrifices some of the "filling," but you might be able to get creative with some other characters to get the best of both worlds.

**Review by spenserblack — COMMENTED** [review]
I like the shading that you've simulated on the logo!

File: languages.yaml (hunk 1 above)
Comment (progDes007) [reply languages.yaml]: Done

File: languages.yaml [hunk 2]
```diff
@@ -1071,6 +1071,38 @@ Hcl:
...
+  colors:
+    ansi:
+      - green
+      - green
+      - white
+    hex:
+      - "#00FF00"
+      - "#006600"
+      - "#FFFFFF"
+    chip: "#AACE60"
```
Comment (o2sh) [inline languages.yaml:1105]: Hi @progDes007, the teapot looks great! 😄 Although I understand your reasoning behind using it as an emblem for this language, given the absence of an official logo for HLSL, it still feels somewhat arbitrary to me. Personally, I'm not particularly fond of the text below the teapot either, especially considering that we already use a similar design in the glsl logo. Considering HLSL is a Microsoft language, I'd like to offer a suggestion:

image

```Hlsl:
  type: programming
  ascii: |
    {0}████████████████  {1}████████████████
    {0}█████  ██  █████  {1}█████  █████████
    {0}█████  ██  █████  {1}█████  █████████
    {0}█████      █████  {1}█████  █████████
    {0}█████  ██  █████  {1}█████  █████████
    {0}█████  ██  █████  {1}█████      █████
    {0}████████████████  {1}████████████████

    {2}████████████████  {3}████████████████
    {2}█████      █████  {3}█████  █████████
    {2}█████  █████████  {3}█████  █████████
    {2}█████      █████  {3}█████  █████████
    {2}█████████  █████  {3}█████  █████████
    {2}█████      █████  {3}█████      █████
    {2}████████████████  {3}████████████████
  colors:
    ansi:
      - red
      - green
      - blue
      - yellow
    hex:
      - "#F65314"
      - "#7CBB00"
      - "#00A1F1"
      - "#FFBB00"
    chip: "#AACE60"

```

Comment (progDes007) [reply languages.yaml]: @o2sh Added the changes as you suggest

**Review by o2sh — APPROVED** [review]

**Pull request merged**